conda/base/context.py

```diff
@@ -447,12 +447,15 @@ def custom_channels(self):
 
     @property
     def channels(self):
-        # add 'defaults' channel when necessary if --channel is given via the command line
+        # add 'defaults' channel if 'channels' is not defined in configuration files
+        # or environment variables, and 'defaults' is not already given with a `--channel` CLI flag
         if self._argparse_args and 'channel' in self._argparse_args:
             # TODO: it's args.channel right now, not channels
             argparse_channels = tuple(self._argparse_args['channel'] or ())
-            if argparse_channels and argparse_channels == self._channels:
-                return argparse_channels + (DEFAULTS_CHANNEL_NAME,)
+            if not any('channels' in params and location != 'cmd'
```

Comment:
'cmd' or 'cmd_line'?

Reply:
In cmd_line the entry seems to be 'channel' (singular)

Reply:
Yup, you were right.  It didn't matter here, because the argparse argument is named `channel`, not `channels`.  I was hoping to change that at the conda 4.4 transition, although I'm not sure that actually got in.

Either way, if it ever does someday, there are now regression tests that should have us covered :)